File: UngDungDiemDanhNhanVien/DTOs/DangNhapRequest.cs
using System.ComponentModel.DataAnnotations;

namespace UngDungDiemDanhNhanVien.DTOs
{
    public class DangNhapRequest
    {
        [Required(ErrorMessage = "Tên đăng nhập là bắt buộc")]
        public string TenDangNhap { get; set; } = string.Empty;

        [Required(ErrorMessage = "Mật khẩu là bắt buộc")]
        public string MatKhau { get; set; } = string.Empty;
    }
}

File: ung_dung_diem_danh/lib/models/dang_nhap_request.dart
import 'package:equatable/equatable.dart';

class DangNhapRequest extends Equatable {
  final String tenDangNhap;
  final String matKhau;

  const DangNhapRequest({
    required this.tenDangNhap,
    required this.matKhau,
  });

  Map<String, dynamic> toJson() {
    return {
      'tenDangNhap': tenDangNhap,
      'matKhau': matKhau,
    };
  }

  @override
  List<Object?> get props => [tenDangNhap, matKhau];
}

class DangNhapNhanVienRequest extends Equatable {
  final String maNhanVien;
  final String matKhau;

  const DangNhapNhanVienRequest({
    required this.maNhanVien,
    required this.matKhau,
  });

  Map<String, dynamic> toJson() {
    return {
      'maNhanVien': maNhanVien.toString(),
      'matKhau': matKhau,
    };
  }

  @override
  List<Object?> get props => [maNhanVien, matKhau];
}

